### PR TITLE
fix: Make ClassFinder deterministic

### LINKED ISSUE #7556: Support reproducible flow-build-info.json and stats.json
Currently (v14) generated `flow-build-info.json` and `stats.json` entries are randomly reordered during build. In containerized environment it means that every build produces an unique image which creates problems both in development and CD workflows.

This should be also taken into account in relation to #6398

### PR DESCRIPTION
**Pull request by Artur-**
Fixes #7556

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/scanner/ClassFinder.java [hunk 1]
```diff
@@ -70,7 +80,7 @@ public DefaultClassFinder(Set<Class<?>> classes) {
         public DefaultClassFinder(ClassLoader classLoader,
                 Class<?>... classes) {
             this.classLoader = classLoader;
-            this.classes = new HashSet<>();
+            this.classes = new LinkedHashSet<>();
             for (Class<?> clazz : classes) {
                 this.classes.add(clazz);
             }
```
Comment (caalador) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/scanner/ClassFinder.java:86]: Shouldn't this add also be sorted for the classes array.
So something like `Arrays.stream(classes).sorted(Comparator.comparing(Class::getName)).forEach(this.classes::add);`
As there is nothing guaranteeing that the classes is always ordered when given.

**Review by caalador — CHANGES_REQUESTED** [review]

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/scanner/ClassFinder.java (hunk 1 above)
Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/scanner/ClassFinder.java:86]: It is not needed for ClassFinder to be deterministic. For the Set constructor it is needed as the iteration order for the same classes differ when run multiple times. Wouldn't it be expected that the classes are scanned in the order they are given?

Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/scanner/ClassFinder.java:86]: Also it seems this constructor is used only in tests.

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdateImportsTest.java [hunk 2]
```diff
@@ -393,6 +395,45 @@ Logger log() {
...
+        JsonObject fallback = fallBackData.getObject("chunks")
+                .getObject("fallback");
+        JsonArray jsModules = fallback.getArray("jsModules");
+        JsonArray cssImports = fallback.getArray("cssImports");
+
+        String expectedJsModules = "[\"@polymer/e.js\",\"@polymer/D.js\",\"@polymer/c.js\",\"@polymer/b.js\",\"@polymer/a.js\",\"./extra-javascript.js\"]";
+        String expectedCssImports = "[{\"value\":\"./b-css.css\"},{\"include\":\"a-a\",\"value\":\"./a-css.css\"},{\"include\":\"extra-bar\",\"themeFor\":\"extra-foo\",\"value\":\"./extra-css.css\"}]";
+
+        Assert.assertEquals(expectedJsModules, jsModules.toJson());
+        Assert.assertEquals(expectedCssImports, cssImports.toJson());
```
Comment (caalador) [inline flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdateImportsTest.java:434]: Should also assert the tokenFile content for the written data here.

**Review by caalador — CHANGES_REQUESTED** [review]
A more certain place to keep the build file in order would be to order the written imports in `TaskUpdateImports::populateFallbackData`

At least the token file should be asserted in the test

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdateImportsTest.java (hunk 2 above)
Comment (Artur-) [reply flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdateImportsTest.java:434]: That can be done although the token file contains nothing else than the fallback chunk

Comment (Artur-) [reply flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdateImportsTest.java:436]: Added

**Review by caalador — APPROVED** [review]

**Pull request merged**